packages/grafana-schema/src/schema/dashboard/v2alpha0/dashboard.schema.cue

```diff
@@ -0,0 +1,790 @@
+// "Tooltip" for shared crosshair AND shared tooltip.
+DashboardCursorSync: "Off" | "Crosshair" | "Tooltip"
+
+// Links with references to other dashboards or external resources
+DashboardLink: {
+  // Title to display with the link
+  title: string
+  // Link type. Accepted values are dashboards (to refer to another dashboard) and link (to refer to an external resource)
+  // FIXME: The type is generated as `type: DashboardLinkType | dashboardLinkType.Link;` but it should be `type: DashboardLinkType`
+  type: DashboardLinkType
+  // Icon name to be displayed with the link
+  icon: string
+  // Tooltip to display when the user hovers their mouse over it
+  tooltip: string
+  // Link URL. Only required/valid if the type is link
+  url?: string
+  // List of tags to limit the linked dashboards. If empty, all dashboards will be displayed. Only valid if the type is dashboards
+  tags: [...string]
+  // If true, all dashboards links will be displayed in a dropdown. If false, all dashboards links will be displayed side by side. Only valid if the type is dashboards
+  asDropdown: bool | *false
+  // If true, the link will be opened in a new tab
+  targetBlank: bool | *false
+  // If true, includes current template variables values in the link as query params
+  includeVars: bool | *false
+  // If true, includes current time range in the link as query params
+  keepTime: bool | *false
+}
+
+DataSourceRef: {
+  // The plugin type-id
+  type?: string
+
+  // Specific datasource instance
+  uid?: string
+}
+
+// Transformations allow to manipulate data returned by a query before the system applies a visualization.
+// Using transformations you can: rename fields, join time series data, perform mathematical operations across queries,
+// use the output of one transformation as the input to another transformation, etc.
+DataTransformerConfig: {
+  // Unique identifier of transformer
+  id: string
+  // Disabled transformations are skipped
+  disabled?: bool
+  // Optional frame matcher. When missing it will be applied to all results
+  filter?: MatcherConfig
+  // Where to pull DataFrames from as input to transformation
+  topic?: "series" | "annotations" | "alertStates" // replaced with common.DataTopic
+  // Options to be passed to the transformer
+  // Valid options depend on the transformer id
+  options: _
+}
+
+// The data model used in Grafana, namely the data frame, is a columnar-oriented table structure that unifies both time series and table query results.
+// Each column within this structure is called a field. A field can represent a single time series or table column.
+// Field options allow you to change how the data is displayed in your visualizations.
+FieldConfigSource: {
+  // Defaults are the options applied to all fields.
+  defaults: FieldConfig
+  // Overrides are the options applied to specific fields overriding the defaults.
+  overrides: [...{
+    matcher: MatcherConfig
+    properties: [...DynamicConfigValue]
+  }]
+}
```

Comment:
Is there any chance we could re-use existing types like this? The ones that are not really affected by the schema changes? I'm not convinced we should re-implement types that are not affected by the schema work.

Reply:
@ivanortegaalba  - can you add this as a work item to the epic so that it doesn't slip?